cli/command/manifest/push.go

```diff
@@ -271,3 +307,47 @@ func mountBlobs(ctx context.Context, client registryclient.RegistryClient, ref r
 	}
 	return nil
 }
+
+func annotateListFromYaml(cli command.Cli, yaml yamlManifestList) error {
+
+	for _, manifest := range yaml.Manifests {
+		opts := annotateOptions{
+			target:     yaml.Image,
+			image:      manifest.Image,
+			variant:    manifest.Platform.Variant,
+			os:         manifest.Platform.OS,
+			arch:       manifest.Platform.Architecture,
+			osFeatures: manifest.Platform.OSFeatures,
+		}
+		if err := runManifestAnnotate(cli, opts); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func createListFromYaml(cli command.Cli, yamlInput yamlManifestList, insecureRegistry bool) error {
+
+	var args []string
+	args = append(args, yamlInput.Image)
+	for _, manifest := range yamlInput.Manifests {
+		args = append(args, manifest.Image)
+	}
+	return createManifestList(cli, args, createOpts{amend: false, insecure: insecureRegistry})
+}
+
+func getYamlManifestList(dockerCli command.Cli, yamlFile string) (yamlManifestList, error) {
+	var yamlInput yamlManifestList
+
+	if _, err := os.Stat(yamlFile); err != nil {
+		return yamlManifestList{}, err
+	}
```

Comment:
Is this Stat necessary? I think you'd get the same error from `ReadFile()`.

Reply:
nope, thanks!